### PR TITLE
feat(update-index): automate adding release notes to index

### PR DESCRIPTION
**Pull request by tolzhabayev**
**Why?**
Currently, we have to update `_index.md` file manually when creating a changelog file. This PR automates it.

**Additional changes**
- Remove `nock` as it is not really used anymore
- Make unit tests run silently per default

Fixes: #13

### CODE REVIEW

**Review by hugohaggmark — COMMENTED** [review]
What happens if we run UpdateChangelog action for a previous release, let's say 8.3.0. Will this still work? I think we should add that as a test case either way.

**Comment by tolzhabayev** [comment]
> What happens if we run UpdateChangelog action for a previous release, let's say 8.3.0. Will this still work? I think we should add that as a test case either way.

good point, I will look into making it not "forward-only".

**Comment by marefr** [comment]
I was under the impression that this has been sort of supported earlier, but I might be confused/remember incorrectly. I'll have a look at the changes :+1:

File: update-changelog/testdata/_index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++
+
+# Release notes
+
+Here you can find detailed release notes that list everything that is included in every release as well as notices
+about deprecations, breaking changes as well as changes that relate to plugin development.
+
+- [Release notes for 8.3.2]({{< relref "release-notes-8-3-2" >}})
+- [Release notes for 8.3.1]({{< relref "release-notes-8-3-1" >}})
+- [Release notes for 8.3.0]({{< relref "release-notes-8-3-0" >}})
```
Comment (marefr) [inline update-changelog/testdata/_index.md:13]: Based on earlier comment in PR. With this update adding a test for generating changelog for 8.2.2 would be nice and it should be inserted between 8.3.0 and 8.2.1 :) Maybe can make use of markers as we do with the changelog.md file, i.e. `<!-- 8.3.3 START -->` and `<!-- 8.3.3 END -->` etc, so you don't have to parse the markdown. 

```suggestion
- [Release notes for 8.3.2]({{< relref "release-notes-8-3-2" >}})
- [Release notes for 8.3.1]({{< relref "release-notes-8-3-1" >}})
- [Release notes for 8.3.0]({{< relref "release-notes-8-3-0" >}})
- [Release notes for 8.2.1]({{< relref "release-notes-8-2-1" >}})
- [Release notes for 8.2.0]({{< relref "release-notes-8-2-0" >}})
```

**Review by marefr — CHANGES_REQUESTED** [review]
See comment

**Comment by tolzhabayev** [comment]
@marefr @hugohaggmark please re-review, it should now add it into the proper position for any version.

**Review by marefr — APPROVED** [review]
Great LGTM

**Review by hugohaggmark — APPROVED** [review]
Great work on this, so awesome that you added all those tests 🎉

**Pull request merged**